investor/LoanFilter.py

```diff
@@ -23,16 +24,11 @@ def __str__(self):
 		pass
 
 	@abstractmethod
-	def _eval(self, loan):
+	def _eval(self, loan, block):
 		pass
 
-	def apply(self, loan):
-		if self._eval(loan):
-			self.pass_count += 1
-			return True
-		else:
-			self.fail_count += 1
-			return False
+	def apply(self, loan, block=True):
+		return self._eval(loan, block)
```

Comment:
How about spawning a second thread from apply() on a 1 second timer, i.e. wait 1sec before executing, that iterates through a queue of results populated by apply() and tallies up the pass/fail?


From what I read it's difficult to prioritize threads in python, otherwise I would suggest creating a low priority thread to do it.